packages/lit-dev-content/site/docs/components/styles.md

```diff
@@ -74,6 +75,12 @@ Using an array of tagged template literals, a component can inherit the styles f
 
 {% playground-ide "docs/components/style/superstyles" %}
 
+<div class="alert alert-info">
+
+**Using super.styles**. You can also use `super.styles` to reference the superclass's styles property, but TypeScript doesn't convert this correctly when compiling for ES5. Explicitly referencing the superclass, as shown in the example, avoids this issue.
```

Comment:
Maybe clarify that this does work for ES2015 and above, though. Also I think we should show the `super.styles` approach in the examples. (@AndrewJakubowicz found that it works fine in TypeScript 4.4, we just need to update the Playground's TypeScript version).

Reply:
Oh, cool. Yeah, I agree that showing the `super.styles` approach as the default is preferable. Is there a bug for updating the Playground typescript version?

Reply:
https://github.com/google/playground-elements/pull/217 to release with the new version. After this I'll send a PR to update lit.dev. Should be able to get that in before this PR merges.

Reply:
https://github.com/lit/lit.dev/pull/537

Reply:
Huh, bumping the TypeScript version fixed the type error, so there's no longer a squiggly when doing `super.styles`, but it didn't fix the output, which is still invalid note the `(void 0).styles`:

```ts
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { css } from './node_modules/lit@2.0.0/index.js';
import { customElement } from './node_modules/lit@2.0.0/decorators.js';
import { SuperElement } from './super-element.js';
let MyElement = class MyElement extends SuperElement {
};
MyElement.styles = [
    (void 0).styles,
    css `div {
      color: red;
    }`
];
MyElement = __decorate([
    customElement('my-element')
], MyElement);
export { MyElement };
```

I wonder if we need to tweak the compiler options (https://github.com/google/playground-elements/blob/05538da0ee6fe13b3345d89b805dd06b2b8f30b3/src/typescript-worker/language-service-context.ts#L9)

Reply:
I still see an error in 4.4 and 4.5 when targeting ES5, but the ES2015 output does look correct. Oddly, I'm not seeing the `(void 0)` emit that you are @aomarks.

https://www.typescriptlang.org/play?target=1&ts=4.4.3#code/MYGwhgzhAECC0G8BQ1oQC5nQS2G9AniAKYwC80A5GJQNxIC+SSokMAQtMQB7rEB2AExjxkqDFlz4ipaBQgBXAA7EATgDoMMmAGoqAIzqMgA

I guess showing `super.styles` by default is ok... I just worry that code like that might not be able to be down-leveled by Babel in the case of an app that needs to target ES5 and is using 3rd party dependencies.

Reply:
@justinfagnani Interesting. It only happens when there is a decorator on the subclass:

https://www.typescriptlang.org/play?emitDecoratorMetadata=false&target=7&jsx=0#code/MYGwhgzhAECC0G8BQ1oQC5nQS2G9AniAKYwC80A5GJQNxIC+SSAAsAK4YD2AtgKIkexAHboAFAEokoSDABC0YgA90IgCYx4yVBiy58RUtAoR2AB2IAnAHQZDMANRUARnUZA

Reply:
Did we get a resolution for this?

Reply:
Oh I see, the latest update seems fine (though maybe we should note it's also broken for ES2015+ with a class decorator, not just ES5?) I think we should file a typescript bug?

Reply:
I think under those circumstances we should just recommend against using it in TypeScript. If someone uses it and subsequently decides to add an IE11 build or to adopt decorators, their code will break and that seems bad.

Reply:
... especially since we invariably show TypeScript with decorators, so the only circumstance where it works is off the well-lit path.

Reply:
Made a [repro](https://www.typescriptlang.org/play?target=99#code/KYDwDg9gTgLgBAbwL4G4BQaD0Aqba7ZwAqAFgJYDOcUwYUEAJgK4DGwVAhnAEZMDmcAO5kYJOBRgcYZFuJgBPADbs4HAHYM4DYC2hToFAHT5CpYHDJqWiphTIQ1cCADNVWnXpjQ4LDrZWi5gAGFExgwFCGEkrsQXBePOaWulA0LDCK8nDAALYiMMCaHFRBABQAbhBkmgAMAJRBxgSYaM5MVtIO2SAcOWDKpelQAFzx8uEucADKYREAworFFHWINDBMUI5DKEgY1kvTs1ALBwj4clIycjFUALxwANoA5NHKTwC66Lto+xRUM+EoFMFMoAOrQADWVFABQ0-yOJz+iHOEkusleKnuD0MONCgKiIPYnzgmEwcDm0DS8Fy+UKPkYwFGQQBEWBMXBUChBJucCxOMMACVgM5lOlDHxgDBSgB9bgAGjgACIMRRFQrpRw6p8gmhvgABUC9frAH6LJEAIXoEOAjhhNoY8MBiKoZzgF2k6MJd0e-LxEW5ygoxNJcAAklZKTpqXkYAVNLptEzLRBrWoA5ifTiKlVanV00GUDrdkA). Can look for an open issue in TypeScript or open a new one.

Reply:
Another maybe interesting note, if you check `useDefineForClassFields` then the (void 0) bug no longer occurs.